File: Finanzuebersicht.Frontend.Web.Admin/src/app/model/accounting/categories/dtos/i-category-detail.ts
import { IAccountingEntry } from 'src/app/model/accounting/accounting-entries/dtos/i-accounting-entry';
import { ICategory } from 'src/app/model/accounting/categories/dtos/i-category';
import { ICategorySearchTerm } from 'src/app/model/accounting/category-search-terms/dtos/i-category-search-term';

export interface ICategoryDetail {
    id: string;
    accountingEntries: IAccountingEntry[];
    childCategories: ICategory[];
    superCategory: ICategory;
    categorySearchTerms: ICategorySearchTerm[];
    title: string;
    color: string;
}

File: Finanzuebersicht.CodeGeneration/2-Post-Contractor/Backend/Contract/Logic/Modules/Accounting/Categories/DTOs/ICategoryDetail.cs
using Finanzuebersicht.Backend.Generated.Contract.Logic.Modules.Accounting.AccountingEntries;
using Finanzuebersicht.Backend.Generated.Contract.Logic.Modules.Accounting.Categories;
using Finanzuebersicht.Backend.Generated.Contract.Logic.Modules.Accounting.CategorySearchTerms;
using System;
using System.Collections.Generic;

namespace Finanzuebersicht.Backend.Generated.Contract.Logic.Modules.Accounting.Categories
{
    public interface ICategoryDetail
    {
        Guid Id { get; set; }

        IEnumerable<IAccountingEntry> AccountingEntries { get; set; }

        IEnumerable<ICategory> ChildCategories { get; set; }

        ICategory SuperCategory { get; set; }

        IEnumerable<ICategorySearchTerm> CategorySearchTerms { get; set; }

        string Title { get; set; }

        string Color { get; set; }
    }
}